src/main/java/com/jcs/sbs/model/DescribeVolumesRequest.java

```diff
@@ -5,181 +5,202 @@
-        return this.maxResults;
-    }
-    
-    /**
-     * @param maxResults
-     * @return
-     */
-    public DescribeVolumesRequest withMaxResults(Integer maxResults) {
-        setMaxResults(maxResults);
-        return this;
-    }
-
-    @Override
-    public String toString() {
-        StringBuilder sb = new StringBuilder();
-        sb.append("{");
-        if (getVolumeIds() != null)
-            sb.append("VolumeIds: " + getVolumeIds() + ",");
-        if (getNextToken() != null)
-            sb.append("NextToken: " + getNextToken() + ",");
-        if (getMaxResults() != null)
-            sb.append("MaxResults: " + getMaxResults());
-        sb.append("}");
-        return sb.toString();
-    }
-
-    @Override
-    public boolean equals(Object obj) {
-        if (this == obj)
-            return true;
-        if (obj == null)
-            return false;
-
-        if (obj instanceof DescribeVolumesRequest == false)
-            return false;
-        DescribeVolumesRequest other = (DescribeVolumesRequest) obj;
-        if (other.getVolumeIds() == null ^ this.getVolumeIds() == null)
-            return false;
-        if (other.getVolumeIds() != null
-                && other.getVolumeIds().equals(this.getVolumeIds()) == false)
-            return false;
-        if (other.getNextToken() == null ^ this.getNextToken() == null)
-            return false;
-        if (other.getNextToken() != null
-                && other.getNextToken().equals(this.getNextToken()) == false)
-            return false;
-        if (other.getMaxResults() == null ^ this.getMaxResults() == null)
-            return false;
-        if (other.getMaxResults() != null
-                && other.getMaxResults().equals(this.getMaxResults()) == false)
-            return false;
-        return true;
-    }
-
-    @Override
-    public int hashCode() {
-        final int prime = 31;
-        int hashCode = 1;
-
-        hashCode = prime * hashCode
-                + ((getVolumeIds() == null) ? 0 : getVolumeIds().hashCode());
-        hashCode = prime * hashCode
-                + ((getNextToken() == null) ? 0 : getNextToken().hashCode());
-        hashCode = prime * hashCode
-                + ((getMaxResults() == null) ? 0 : getMaxResults().hashCode());
-        return hashCode;
-    }
-
-    @Override
-    public DescribeVolumesRequest clone() throws CloneNotSupportedException {
-        return (DescribeVolumesRequest) super.clone();
-    }
+
+	private String nextToken;
+
+	private Integer maxResults;
+
+	private Boolean detail;
+
+	public DescribeVolumesRequest() {
+	}
+
+	/**
+	 * @param volumeIds
+	 */
+	public DescribeVolumesRequest(List<String> volumeIds) {
+		setVolumeIds(volumeIds);
+	}
+
+	/**
+	 * @return
+	 */
+	public List<String> getVolumeIds() {
+		if (volumeIds == null) {
+			volumeIds = new ArrayList<String>();
+		}
+		return volumeIds;
+	}
+
+	/**
+	 * @param volumeIds
+	 */
+	public void setVolumeIds(Collection<String> volumeIds) {
```

Comment:
BTW I followed best practices and hence named like this. Please also see the accepted answer in this link:
http://stackoverflow.com/questions/991757/best-practice-for-parameter-naming-in-java-constructors-and-simple-setters

If you will disagree, then I will change the names.